app-frontend/src/app/components/vectors/vectorImportModal/vectorImportModal.module.js

```diff
@@ -0,0 +1,246 @@
+    }
+
+    allowPrevious() {
+        return this.currentStep.allowPrevious && this.currentStep.allowPrevious();
+    }
+
+    handlePrevious() {
+        this.setCurrentStep(this.getStep(this.currentStep.previous()));
+    }
+
+    hasNext() {
+        return this.currentStep.next && this.currentStep.next();
+    }
+
+    allowNext() {
+        return this.currentStep.allowNext && this.currentStep.allowNext();
+    }
+
+    handleNext() {
+        this.setCurrentStep(this.getStep(this.currentStep.next()));
+    }
+
+    allowDone() {
+        return this.currentStep.allowDone && this.currentStep.allowDone();
+    }
+
+    handleDone() {
+        this.close();
+    }
+
+    currentStepIs(step) {
+        if (step.constructor === Array) {
+            return step.reduce((acc, cs) => {
+                return acc || this.currentStepIs(cs);
+            }, false);
+        }
+        return this.currentStep.name === step;
+    }
+
+    setCurrentStep(step) {
+        if (this.currentStep && this.currentStep.onExit) {
+            this.currentStep.onExit();
+        }
+        this.currentStep = step;
+        if (this.currentStep.onEnter) {
+            this.currentStep.onEnter();
+        }
+    }
+
+    preventInterruptions() {
+        if (!this.closeCanceller) {
+            this.closeCanceller = this.$scope.$on('modal.closing', (e) => {
+                e.preventDefault();
+            });
+            this.locationChangeCanceller = this.$scope.$on('$locationChangeStart', (event) => {
+                event.preventDefault();
+            });
+        }
+    }
+
+    allowInterruptions() {
+        if (this.closeCanceller) {
+            this.closeCanceller();
+            delete this.closeCanceller;
+        }
+        if (this.locationChangeCanceller) {
+            this.locationChangeCanceller();
+            delete this.locationChangeCanceller;
+        }
+    }
+
+    // SELECT_FILE
+
+    verifyFileCount() {
+        return Boolean(this.selectedFile);
+    }
+
+    fileSelected(file) {
+        this.selectedFile = file;
+    }
+
+    getFileSize() {
+        if (this.selectedFile.length) {
+            return this.selectedFile.reduce((acc, f) => {
+                return acc + f.size;
+            }, 0);
+        }
+        return 0;
+    }
+
+    removeFile() {
+        delete this.selectedFile;
+    }
+
+    // UPLOADING_FILE
+
+    startUpload() {
+        this.preventInterruptions();
+        if (this.shapeName && this.shapeName.length) {
+            this.selectedFile = this.Upload.rename(this.selectedFile, this.shapeName);
+        }
+        this.upload = this.Upload.upload({
+            url: `${BUILDCONFIG.API_HOST}/api/shapes/upload`,
+            data: {
+                name: this.selectedFile
+            },
+            headers: {
+                Authorization: `Bearer ${this.authService.token()}`
+            }
+        });
+        this.upload.then((resp) => {
```

Comment:
Uploading a file that isn't a valid zipped shapefile causes the upload to appear stuck at 100% which means the user can't close the modal in anyway. There should probably be some escape hatch and an indication that things didn't work.

Reply:
🤔 yup